pkg/operator/controllers/genevalogging/genevalogging_controller.go

```diff
@@ -34,66 +36,104 @@ const (
 
 // Reconciler reconciles a Cluster object
 type Reconciler struct {
-	log *logrus.Entry
+	log  *logrus.Entry
+	role string
 
 	dh     dynamichelper.Interface
 	client client.Client
 }
 
-func NewReconciler(log *logrus.Entry, client client.Client, dh dynamichelper.Interface) *Reconciler {
+func NewReconciler(log *logrus.Entry, client client.Client, dh dynamichelper.Interface, role string) *Reconciler {
 	return &Reconciler{
-		log: log,
+		log:  log,
+		role: role,
 
 		dh:     dh,
 		client: client,
 	}
 }
 
-// Reconcile the genevalogging deployment.
-func (r *Reconciler) Reconcile(ctx context.Context, request ctrl.Request) (ctrl.Result, error) {
-	instance := &arov1alpha1.Cluster{}
-	err := r.client.Get(ctx, types.NamespacedName{Name: arov1alpha1.SingletonClusterName}, instance)
+func (r *Reconciler) ensureResources(ctx context.Context, instance *arov1alpha1.Cluster) error {
+	operatorSecret := &corev1.Secret{}
+	err := r.client.Get(ctx, types.NamespacedName{Namespace: operator.Namespace, Name: operator.SecretName}, operatorSecret)
 	if err != nil {
-		return reconcile.Result{}, err
+		return err
 	}
 
-	if !instance.Spec.OperatorFlags.GetSimpleBoolean(controllerEnabled) {
-		r.log.Debug("controller is disabled")
-		return reconcile.Result{}, nil
+	resources, err := r.resources(ctx, instance, operatorSecret.Data[GenevaCertName], operatorSecret.Data[GenevaKeyName])
+	if err != nil {
+		return err
 	}
 
-	r.log.Debug("running")
-	operatorSecret := &corev1.Secret{}
-	err = r.client.Get(ctx, types.NamespacedName{Namespace: operator.Namespace, Name: operator.SecretName}, operatorSecret)
+	err = dynamichelper.SetControllerReferences(resources, instance)
 	if err != nil {
-		return reconcile.Result{}, err
+		return err
 	}
 
-	resources, err := r.resources(ctx, instance, operatorSecret.Data[GenevaCertName], operatorSecret.Data[GenevaKeyName])
+	err = dynamichelper.Prepare(resources)
 	if err != nil {
-		r.log.Error(err)
-		return reconcile.Result{}, err
+		return err
 	}
 
-	err = dynamichelper.SetControllerReferences(resources, instance)
+	err = r.dh.Ensure(ctx, resources...)
 	if err != nil {
-		r.log.Error(err)
-		return reconcile.Result{}, err
+		return err
 	}
 
-	err = dynamichelper.Prepare(resources)
+	return nil
+}
+
+// Reconcile the genevalogging deployment.
+func (r *Reconciler) Reconcile(ctx context.Context, request ctrl.Request) (ctrl.Result, error) {
+	instance := &arov1alpha1.Cluster{}
+	err := r.client.Get(ctx, types.NamespacedName{Name: arov1alpha1.SingletonClusterName}, instance)
 	if err != nil {
 		r.log.Error(err)
 		return reconcile.Result{}, err
 	}
 
-	err = r.dh.Ensure(ctx, resources...)
+	if !instance.Spec.OperatorFlags.GetSimpleBoolean(controllerEnabled) {
+		r.log.Debug("controller is disabled")
+		return r.reconcileDisabled(ctx)
+	}
+
+	r.log.Debug("running")
+
+	reconcileErr := r.ensureResources(ctx, instance)
+
+	err = conditions.SetCondition(ctx, r.client, r.condition(reconcileErr), r.role)
```

Comment:
I think for consistency, the way to implement this would be to use the helpers/interfaces Tanmay wrote that we've been using for all other controllers, which will eliminate some duplicate code here. We'd be looking to have our Reconciler implement the `AROController`, and then use the `r.SetDegraded(ctx, err)` helper here.

ref:
https://github.com/Azure/ARO-RP/blob/master/pkg/operator/controllers/base/aro_controller.go